src/node/interfaces.cpp

```diff
@@ -345,6 +345,7 @@ bool FillBlock(const CBlockIndex* index, const FoundBlock& block, UniqueLock<Rec
         REVERSE_LOCK(lock);
         if (!ReadBlockFromDisk(*block.m_data, index, Params().GetConsensus())) block.m_data->SetNull();
     }
+    block.found = true;
```

Comment:
Should `found` be set to false in the first line (`if (!index)`)?

Reply:
> Should `found` be set to false in the first line (`if (!index)`)?

The `found` member is just set to `false` in the constructor, and set to `true` when the block is found. I think setting it other places would just add complexity and possible confusion (and I guess be less efficient, not that efficiency matters).

Implementation here is also consistent with the current `FoundBlock` design that only sets values when a block is found, and doesn't set anything when a block isn't found.